### ISSUE TITLE
Feature Request: add user configurable coverart name 

### ISSUE THREAD

**Issue by snake3ca** [MusicPlayerDaemon/MPD#301]
Please add the ability to configure the artwork file name so you dont have to have the artwork named cover.jpg.  Or you could just have it look for cover.jpg or folder.jpg since many other media players / taggers use folder.jpg for artwork.  Im referring to MPDs future ability to send cover art in binary form to clients.

**Comment by ondras** [comment]
+1 for this. My music database uses `Cover.*` (capitalized), which is obviously not working with the current implementation.

**Comment by adelias** [comment]
I agree, it would be nice if this were configurable so I wouldn't have to edit FileCommands.cxx every time.

**Comment by muesli4** [comment]
In [mpd-touch-screen-gui](https://github.com/muesli4/mpd-touch-screen-gui), I am using the following for cover detection:
```
    cover:
    {
        # Add this line for cover support from the filesystem.
        directory = "/home/alarm/Music/"

        # All combinations are tried in the given order to find covers. If none
        # of these matches, the cover replacement is displayed.
        extensions = ["png", "jpeg", "jpg"]
        names = ["front", "cover", "back"]
    }
```
Would you mind it being implemented in a similar way?  I like the idea to have some priority so you can have several files in the directory and it will pick the right one according to the user's specification.

**Comment by snake3ca** [comment]
That would work perfect for me.

**Comment by nguillaumin** [comment]
Related suggestions:
1. Support some sort of template system for users who name their cover file after the album name, for example `/music/Metallica - Master of Puppet/Master of Puppets.jpg` . I'd imagine it would be configured with something like `art_filename = %album` or similar.
2. Implement a fallback where the first image found in the folder is used, rather than expecting a specific filename. This would probably work in most cases and would cater for all "special" cases.

In fact maybe 2. would be good enough for most cases and quick & easy to implement...

**Comment by muesli4** [comment]
@nguillaumin apart from being redundant this scheme is very prone to errors (e.g., due to bad tags or filenames) and your solution is not as flexible as the solution that I proposed.  I don't have anything against allowing templates in it but it makes it slightly more complicated.  For example, what if the relevant tags are missing?  What if the tags contain invalid characters for the file system?  Obviously, one would have to do the same check for presets.

**Comment by arbv** [comment]
Please make the cover art file names configurable (please notice the plural). For example, my DeadBeef configuration looks for :
* cover.jpg, cover.png;
* Cover.jpg, Cover.png;
* folder.jpg, folder.png
* front.jpg, front.png
* Front.jpg, Front.png
* album.jpg, album.png
* Album.jpg, Album.png
* artwork.jpg, artwork.png

That is, making the list of artwork filenames to try configurable via a configuration option would make it possible to make MPD server anybody's needs. It is what, e.g. DeadBeef or FooBar 2000 do.

**Comment by arbv** [comment]
To be more precise, I am on @muesli4 side here, although I would drop a separate configuration option for extensions, just making the list of filenames to try should be enough.

**Comment by muesli4** [comment]
@arbv well, the cartesian product is multiplicative in size.  For 3 each you have to specify only 6 this way vs. 9.  For anything more it gets a hassle quickly.  And since it is really not hard to implement, I really don't see why one should not go the more flexible way.

Then of course there are people that think too much configurability is harmful or too many configuration options are confusing.  But I find this hostility against configurable software and always going the way most people use it a scourge of our time (i.e. mainstreaming).  If I cannot use software the way I want, depending on severity it will make me quit using it.

**Comment by arbv** [comment]
@muesli4 I agree with you at the highest level. To be fair, I would be glad if ability to specify multiple artwork file names would have been implemented  in any way.

I have proposed a variant where only one option is needed just in case asking for multiple options is too much.

**Comment by dgcampea** [comment]
> @arbv well, the cartesian product is multiplicative in size. 

The solution to this is to specify the desired covers as a regex expression (https://github.com/MusicPlayerDaemon/MPD/issues/1493).

**Comment by muesli4** [comment]
> > @arbv well, the cartesian product is multiplicative in size.
> 
> The solution to this is to specify the desired covers as a regex expression (#1493).

Fair enough.  But you have to make sure that the match on the file names is order-dependent, in case there are several files that match.

**Comment by bimac** [comment]
I mean, yeah, regex would be nice, but in the meantime exposing the `art_names` array as a user configurable variable would be a nice and safe workaround, no?

https://github.com/MusicPlayerDaemon/MPD/blob/cb72bc5bd17eddeb8bacd05ec5a3966f250014ac/src/command/FileCommands.cxx#L141-L145
